v1/errors/reporter.go

```diff
@@ -0,0 +1,97 @@
+// errors implements an errors/crashes reporter.
+package errors
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/getsentry/sentry-go"
+	"gopkg.in/inconshreveable/log15.v2"
+)
+
+// Reporter represents an errors reporter instance.
+type Reporter struct {
+	sentry *sentry.Client
+
+	config *Config
+}
+
+// New returns a new errors reporter instance.
+func New(config *Config) (*Reporter, error) {
+	var (
+		reporter Reporter
+		err      error
+	)
+
+	if config == nil {
+		return nil, nil
+	}
+
+	if err := config.validate(); err != nil {
+		return nil, err
+	}
+	reporter.config = config
+
+	if reporter.sentry, err = sentry.NewClient(sentry.ClientOptions{
```

Comment:
(untested)
```
	c, err := sentry.NewClient(sentry.ClientOptions{
		Dsn:              config.DSN,
		AttachStacktrace: true,
	})
	
	if err != nil {
		return nil, err
	}
	
	reporter.sentry = c
```

Reply:
@jeromer can we agree on this middle ground? 😉

```go
	reporter.sentry, err = sentry.NewClient(sentry.ClientOptions{
		Dsn:              config.DSN,
		AttachStacktrace: true,
	})
	if err != nil {
		return nil, err
	}
```

Reply:
we can ;)